src/other/utils.jl

```diff
@@ -477,3 +477,16 @@ function _findall(B::BitVector)::Union{UnitRange{Int}, Vector{Int}}
     end
     @assert false "should not be reached"
 end
+
+# Creates a vector like `vect` without the missing type and copies only data at indices in `inds`
+# Fuses two operations that were previously done by getindex + disallowmissing
+# unsafe because it does not check for `inds` values to be not missing (ensured by the caller)
+# Called by `_getindex_disallowmissing(df,...)`
+function _getindex_disallowmissing(vect::AbstractVector{T}, inds::AbstractVector{<:Integer}) where {T}
+    output = Vector{nonmissingtype(T)}(undef, length(inds))
+    last_ind = 0
+    for i in eachindex(inds)
+        output[last_ind+=1] = vect[inds[i]]
```

Comment:
better split out `last_ind += 1` as an operation before. Probably also `@inbounds` here might make sense.

Reply:
Done. I've not added inbounds, as it was called in the calling function already, but I've added it here as well.